Awesome books: with ES6

andrianarivo opened the pull request:
The things I did in this PR:

1. Properly configured Linters (No Errors)
2. Used Github flow with branches
3. Documented my work in a professional way
4. Used ES6 syntax
5. Divided code into ES6 modules 
6. Used Luxon to display the date

andrianarivo committed:
build awesome books with es6 modules

andrianarivo committed:
Build date time with luxon

LionRouge1 commented on modules/BookStore.js:
```diff
@@ -0,0 +1,49 @@
+export default class {
```
- [ ] Kindly add a name to this class, so that it can be imported in another script

LionRouge1 commented on modules/BookStore.js:
```diff
@@ -0,0 +1,49 @@
+export default class {
+  constructor() {
+    this.booksArray = [];
+  }
+
+  // Get data from localstorage
+  getData() {
+    this.booksArray = JSON.parse(localStorage.getItem('books')) || [];
+  }
```
[OPTIONAL]
- [ ] Nice work initializing the `this.booksArray` :clap: However it would be nice to do it in the `constructor` and remove the function `getData`. Your `constructor` could look like this:
```sh
constructor() {
    this.booksArray = JSON.parse(localStorage.getItem('books')) || [];
}
```
That will make your code clean and DRY :muscle:

andrianarivo replied:
I will need the `getData` function somewhere else

LionRouge1 commented on index.js:
```diff
@@ -62,31 +13,20 @@ const listOfBooks = () => {
             <h1 class="text-center">All Awesome Books</h1>
             <ul class="books w-50 mx-auto list-unstyled my-5 list-group border-3 border-dark"></ul>
         </div>`;
```
[OPTIONAL]
- [ ] Kindly avoid DOM manipulation as much as possible because it is costly. Therefore I will suggest you this HTML code directly in the `index.html`

LionRouge1 commented on index.js:
```diff
@@ -62,31 +13,20 @@ const listOfBooks = () => {
-  }
-});
-
 navLinks.forEach((navLink) => {
   navLink.addEventListener('click', (e) => {
     navLinks.forEach((navLink) => {
       navLink.classList.remove('text-primary');
     });
     e.target.classList.add('text-primary');
     mainContainer.innerHTML = '';
+    const addBtn = document.querySelector('#addBtn');
     switch (e.target.id) {
       case 'list':
         listOfBooks();
```
- [ ] This block of code (lines 22 to 110) is too long and confusing. It would be nice to break it into small functions to make your code clean.

LionRouge1 commented on modules/BookStore.js:
- [ ] Nice work dividing your code into modules :clap: However it would be nice to divide your code into a reasonable modules like `class book` and `class storage`

LionRouge1 reviewed (changes requested):
Hi @andrianarivo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

## Highlights
- Good job creating  a `modules` directory for your `.js` files :+1: 
- Nice work using `arrow` syntax for all your functions :1st_place_medal: 
- All linters passed :heavy_check_mark: 

## Required Changes ♻️
- [ ] The list of books is not displaying and when adding a new book it doesn't save in the storage. Kindly make sure all the functionalities of the app are working as expected as requested in the project requirement.

- [ ] Kindly create `index.css` file for your styles as requested in the project requirement(see image)
![Screenshot from 2023-05-01 10-19-54](https://user-images.githubusercontent.com/91142432/235440048-0077d646-a09b-4031-8855-e51517e9853a.png)

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._function

andrianarivo committed:
more modularize

andrianarivo committed:
fix eslint error

didierganthier reviewed (changes requested):
Hi @andrianarivo ,

🎉 Great job on addressing the feedback from the previous review! You've made some solid progress in implementing the requested changes. However, there are still a few areas that need some attention before we can move on to the next project.

<img src="https://media.giphy.com/media/e56n4Z0mmkxZS/giphy.gif" />

## Highlights
- Used arrow functions ✔️ 
- Divided code into modules ✔️ 
- Integrated luxon.js ✔️ 

## Required Changes 🛠
- [ ] Hey 👋🏽 as requested by the previous reviewer you need to have an `index.css` file in the project structure, I understand that you used Bootstrap but the purpose of this requirement is to ensure that you are able to link `index.css` with `index.js` file. 🔗 Since you're using Bootstrap and you'll have no use for an `index.css` file, feel free to add an empty one to fulfill the requirement. 👍🏽 

- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- N/A

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

andrianarivo committed:
added index.css

andrianarivo commented:
> Hi @andrianarivo ,
> 
> 🎉 Great job on addressing the feedback from the previous review! You've made some solid progress in implementing the requested changes. However, there are still a few areas that need some attention before we can move on to the next project.
> 
> ![](https://camo.githubusercontent.com/c21fff7d889bd167e51cd7aa24eaf53dd99cc831b1de1871e67d0ee4ae176e19/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6535366e345a306d6d6b785a532f67697068792e676966) [ ![68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6535366e345a306d6d6b785a532f67697068792e676966](https://camo.githubusercontent.com/c21fff7d889bd167e51cd7aa24eaf53dd99cc831b1de1871e67d0ee4ae176e19/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6535366e345a306d6d6b785a532f67697068792e676966) ](https://camo.githubusercontent.com/c21fff7d889bd167e51cd7aa24eaf53dd99cc831b1de1871e67d0ee4ae176e19/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6535366e345a306d6d6b785a532f67697068792e676966) [ ](https://camo.githubusercontent.com/c21fff7d889bd167e51cd7aa24eaf53dd99cc831b1de1871e67d0ee4ae176e19/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f6535366e345a306d6d6b785a532f67697068792e676966)
> 
> ## Highlights
> * Used arrow functions ✔️
> * Divided code into modules ✔️
> * Integrated luxon.js ✔️
> 
> ## Required Changes 🛠
> * [ ]  Hey 👋🏽 as requested by the previous reviewer you need to have an `index.css` file in the project structure, I understand that you used Bootstrap but the purpose of this requirement is to ensure that you are able to link `index.css` with `index.js` file. 🔗 Since you're using Bootstrap and you'll have no use for an `index.css` file, feel free to add an empty one to fulfill the requirement. 👍🏽
> * Please review the comments under the review for specific changes that still need to be made.
> 
> ## Optional Suggestions 💡
> * N/A
> 
> 🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.
> 
> **Please note**
> 
> * To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.
> 
> As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
> 
> Cheers and happy coding! 👏👏👏

How is an empty CSS blocking me from achieving the project goal ?

IngDavidTM reviewed (approved):
Hi @andrianarivo  🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 
<img src="https://media3.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif"/>

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
